nemo_curator/modules/fuzzy_dedup.py

```diff
@@ -382,6 +384,131 @@ def __call__(self, dataset: DocumentDataset) -> DocumentDataset:
+        )
+        self.connected_components = ConnectedComponents(
+            cache_dir=self.config.cache_dir,
+            jaccard_pairs_path=os.path.join(
+                self.config.cache_dir, "jaccard_similarity_results.parquet"
+            ),
+            id_column=self.config.id_field,
+            convert_str_ids=False,
+            jaccard_threshold=self.config.jaccard_threshold,
+        )
+
+    def __call__(self, dataset: DocumentDataset):
+        # Minhash + LSH
+        print("Stage1: Starting Minhash + LSH computation")
+        minhashLSH = Sequential([self.minhash, self.lsh])
+        buckets_df = minhashLSH(dataset)
+        print("Stage1: Minhash + LSH complete!")
+
+        # Map buckets to lower cardinality distribution
+        print("Stage2 (False Postive Check): Starting Map_Buckets")
+        ddf_mapped_buckets_w_anchors = self.map_buckets.map_buckets_with_anchors(
+            documents_df=dataset.df, buckets_df=buckets_df.df
+        )
+        mapped_buckets_w_anchors_path = os.path.join(
+            self.config.cache_dir, "anchor_docs_with_bk.parquet"
+        )
+        ddf_mapped_buckets_w_anchors.to_parquet(
+            mapped_buckets_w_anchors_path, write_index=False
+        )
+        print("Stage2 (False Postive Check): Map_Buckets Complete!")
+
+        # Shuffle documents based on mapped buckets
+        print("Stage3 (False Postive Check): Shuffle docs")
+        shuffled_docs_path = os.path.join(
+            self.config.cache_dir, "shuffled_docs.parquet"
+        )
+        self.jaccard_shuffle.shuffle_docs_on_buckets(
+            documents_df=dataset.df,
+            bucket_w_anchors_path=mapped_buckets_w_anchors_path,
+            output_shuffled_docs_path=shuffled_docs_path,
+            bucket_mapping_df_blocksize=256,
+            parts_per_worker=1,
+            bucket_parts_per_worker=8,
+        )
+        print("Stage3 (False Postive Check): Shuffle docs complete!")
+
+        # jaccard comparision within buckets
+        print("Stage4 (False Postive Check): Jaccard Similarity in Buckets")
+        jaccard_pairs_path = os.path.join(
+            self.config.cache_dir, "jaccard_similarity_results.parquet"
+        )
+        jaccard_pairs_df = self.jaccard_compute.jaccard_compute(
+            shuffled_docs_path=shuffled_docs_path
+        )
+        jaccard_pairs_df.to_parquet(
+            jaccard_pairs_path,
+            write_index=False,
+            write_metadata_file=False,
+        )
+        print("Stage4 (False Postive Check): Jaccard Similarity in Buckets Complete!")
+
+        # Connected components across buckets
+        print("Stage5: Connected Components across buckets")
+        cc_path = os.path.join(self.config.cache_dir, "connected_components.parquet")
+        self.connected_components.cc_workflow(cc_path)
+        print("Stage5: Connected Components across buckets complete!")
+        return DocumentDataset(dask_cudf.read_parquet(cc_path, split_row_groups=False))
```

Comment:
Agreed, this will only return the document ID & the group/component they belong to. All documents in the same group/component are near duplicates. I'll document this.